pkg/provider/aci.go

```diff
@@ -1167,15 +1185,16 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*azaci.ContainerProbe,
 			}
 		}
 
-		httpGET = &azaci.ContainerHTTPGet{
+		scheme := azaciv2.Scheme(probe.Handler.HTTPGet.Scheme)
+		httpGET = &azaciv2.ContainerHTTPGet{
 			Port:   &portValue,
 			Path:   &probe.Handler.HTTPGet.Path,
-			Scheme: azaci.Scheme(probe.Handler.HTTPGet.Scheme),
+			Scheme: &scheme,
 		}
 	}
 
-	return &azaci.ContainerProbe{
-		Exec:                exec,
+	return &azaciv2.ContainerProbe{
```

Comment:
This looks like a behavior change. In the old code, if probe.Handler.Exec is nil, containerProbe.Exec will be nil. The new code is not nil.

Reply:
It's the same behavior. The difference is for [SDK v1](https://github.com/Azure/azure-sdk-for-go/blob/v64.1.0/services/containerinstance/mgmt/2021-10-01/containerinstance/models.go#L481), the `containerProbe.Exec` was `*[]string`, while in [SDK v2](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/containerinstance/armcontainerinstance/models.go#L116), the `containerProbe.Exec` is `[]*string`.  So the block https://github.com/virtual-kubelet/azure-aci/pull/422/files#diff-f3b87d76a226c19ab2a6db3f4e284949c846b4746dd03f25c6f5f7430553d1aeR1158-R1166 will manage to have an empty list in case container probe is nil

Reply:
Added UT to double-check all cases.